mypy/main.py

```diff
@@ -102,120 +104,152 @@ def process_options(args: List[str]) -> Tuple[List[BuildSource], Options]:
             module to run as script (or None),
             parsed flags)
     """
-    # TODO: Rewrite using argparse.
+
+    footer = textwrap.dedent(
+        """environment variables:
+        MYPYPATH     additional module search path"""
+    )
+
+    parser = argparse.ArgumentParser(prog='mypy', epilog=footer,
+                                     formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    def parse_version(v):
+        version_components = v.split(".")[0:2]
+        if len(version_components) != 2:
+            raise argparse.ArgumentTypeError(
+                "Invalid python version '{}' (expected format: 'x.y')".format(v))
+        if not all(item.isdigit() for item in version_components):
+            raise argparse.ArgumentTypeError("Found non-digit in python version: '{}'".format(v))
+        return (int(version_components[0]), int(version_components[1]))
+
+    parser.add_argument('-v', '--verbose', action='store_true', help="more verbose messages")
```

Comment:
Can you make this a counter? I've changed things in build.py so that trace() calls print only with `-v -v` while log() prints with `-v`.